sim-lib/src/lib.rs

```diff
@@ -35,15 +40,37 @@ pub struct ActivityDefinition {
     pub amt_msat: u64,
 }
 
-// Phase 2: Event Queue
+#[derive(Debug, Error)]
+pub enum PaymentError {
+    #[error("Get info failed {0}")]
+    GetInfoFailed(String),
+    #[error("Other: {0:?}")]
+    Error(#[from] anyhow::Error),
+}
 
-#[allow(dead_code)]
-pub enum PaymentError {}
+#[derive(Debug, Clone)]
+pub struct NodeInfo {
+    pub pubkey: PublicKey,
+    pub alias: String,
+    // TODO: This should be generic for any node impl
+    pub features: HashMap<u32, Feature>,
```

Comment:
I think this will fit most impls, since `Feature` is just a number?

Reply:
Does this need to map to feature or just a map where we can lookup the u32?

Reply:
I suppose a vec of the `u32`s from get info is sufficient. And we'd have to get this for the `destination` nodes rather than the `origin`s?